hail/src/main/scala/is/hail/cxx/Emit.scala

```diff
@@ -876,60 +880,78 @@ class Emitter(fb: FunctionBuilder, nSpecialArgs: Int, ctx: SparkFunctionContext)
              | ${ rowMajort.setup }
              | ${ shapet.setup }
              | ${ datat.setup }
-             | if (${ rowMajort.m } || ${ shapet.m } || ${ datat.m }) {
+             | ${ shapeTup.define }
+             | if (${ datat.m } || ${ rowMajort.m } || ${ shapeMissing.foldRight("false")((b, m) => s"$b || $m") }) {
```

Comment:
I think this still needs to check `shapet.m` before looking at `shapeMissing`, no?

Reply:
Ah right the whole thing could be missing